src/realm/sync/protocol.hpp

```diff
@@ -263,10 +263,11 @@ struct ProtocolErrorInfo {
         ClientResetNoRecovery,
         MigrateToFLX,
         RevertToPBS,
-        // The RefreshUser/RefreshLocation actions are currently generated internally when the
+        // The RefreshUser/RefreshLocation/LogOutUser actions are currently generated internally when the
         // sync websocket is closed with specific error codes.
         RefreshUser,
         RefreshLocation,
+        LogOutUser,
     };
 
     ProtocolErrorInfo() = default;
```

Comment:
This isn't related to your changes, but should we explicitly set the default value for `server_requests_action` to be `Action::NoAction` - the compiler will use whatever value is at 0, whatever it is...